### ISSUE TITLE
react-birthdays - 

### ISSUE THREAD

**Issue by adumith** [pnp/sp-dev-fx-webparts#2445]
### Disclaimer

Yes

### Sample

react-birthdays

### Author(s)

@smaity @joaojmendes

### Question

First I want to congratulate you for such an excellent job.

I am very new to SharePoint and I don't know how to take this WebPart to my website, do you have a manual that explains the step by step?

I downloaded the package (more than 1GB) and I have VS Code, I am not able to continue I don't know what to do.

Thanks in advance.

**Comment by milanholemans** [comment]
Hi adumith, welcome!
The react-birthdays web part is made in SPFx version 1.8.2, this means that you'll need Node.js version 10.x installed on your computer.
When this is done, the first thing you need to do is installing the `gulp` package on your computer. This can be achieved by running `npm i gulp -g` in your terminal.
Once it is done, navigate to the root of the react-birthdays project. Open your terminal and execute the command `npm i`. This will install all needed packages for this particular project.
After this, you can start building the project. Execute following commands:
```
gulp bundle --ship
gulp package-solution --ship
```
These commands will build and package your project into a single file. You'll find the file in the `./sharepoint/solution` folder. There you'll see a file called `birthdays-anniversary.sppkg`. Upload this file to your SharePoint app catalog and you're done!

Good luck!

**Comment by hugoabernier** [comment]
@adumith did @milanholemans's answer help you resolve your issue? 

I will go ahead and mark this issue as answered, but if you experience any more problems please do not hesitate to create a new issue; we were all new to SharePoint at one point in our careers, so please do not be shy!

**Comment by adumith** [comment]
> Hi adumith, welcome! The react-birthdays web part is made in SPFx version 1.8.2, this means that you'll need Node.js version 10.x installed on your computer. When this is done, the first thing you need to do is installing the `gulp` package on your computer. This can be achieved by running `npm i gulp -g` in your terminal. Once it is done, navigate to the root of the react-birthdays project. Open your terminal and execute the command `npm i`. This will install all needed packages for this particular project. After this, you can start building the project. Execute following commands:
> 
> ```
> gulp bundle --ship
> gulp package-solution --ship
> ```
> 
> These commands will build and package your project into a single file. You'll find the file in the `./sharepoint/solution` folder. There you'll see a file called `birthdays-anniversary.sppkg`. Upload this file to your SharePoint app catalog and you're done!
> 
> Good luck!

Hello!

Thank so much.

I already installed Node
Then I opened VS Code / Terminal I ran this command npm i gulp -g, but I got this error:
_PS C:\Users\ajd1\Documents\Code\Happy Birthday> npm i gulp -g
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
added 349 packages, and audited 350 packages in 13s
8 packages are looking for funding
  run `npm fund` for details
5 high severity vulnerabilities
To address all issues (including breaking changes), run:
  npm audit fix --force
Run `npm audit` for details._

After ran `npm fund` I got happy-birthday@1.0.0 (the folder is empty)

I think I'm totally lost or confused, I don't know how to make the web part work on my website.

I hope you can help me.

**Comment by milanholemans** [comment]
Hi @adumith, the logs you are posting don't look like errors but just warnings. It's quite normal when you install node modules that you get warnings. As long as there are no errors you shouldn't worry about this.
If you run the command `npm list -g --depth=0`, you should see `gulp` somewhere. If this is the case, then `gulp` should be installed correctly.
![image](https://user-images.githubusercontent.com/11723921/159990024-609c9a88-1c2e-440c-b78a-cc49ec2c44d0.png)

Just make sure when you write the command `node -v` that you are on version 10.x.x .

**Comment by adumith** [comment]
> Hi @adumith, the logs you are posting don't look like errors but just warnings. It's quite normal when you install node modules that you get warnings. As long as there are no errors you shouldn't worry about this. If you run the command `npm list -g --depth=0`, you should see `gulp` somewhere. If this is the case, then `gulp` should be installed correctly. ![image](https://user-images.githubusercontent.com/11723921/159990024-609c9a88-1c2e-440c-b78a-cc49ec2c44d0.png)
> 
> Just make sure when you write the command `node -v` that you are on version 10.x.x .

Perfect timing... 

Thank you.

I got this:
```
C:\Users\ajd1\AppData\Roaming\npm
├── gulp-cli@2.3.0
├── gulp@4.0.2
└── npm@8.5.5
```
What next?

**TIA**

**Comment by milanholemans** [comment]
run following commands
```
npm i
gulp bundle --ship
gulp package-solution --ship
```

When this is done, your sppkg file should have been created as mentioned in [this reaction](https://github.com/pnp/sp-dev-fx-webparts/issues/2445#issuecomment-1073127580).

**Comment by adumith** [comment]
> gulp bundle --ship

Thank you,

I got this error:
```
PS C:\Users\ajd1\Documents\Code\Happy Birthday> gulp bundle --ship
[15:03:14] Local modules not found in ~\Documents\Code\Happy Birthday
[15:03:14] Try running: npm install
```

**Comment by milanholemans** [comment]
Did you run `npm install` ?

**Comment by adumith** [comment]
> Did you run `npm install` ?

Hello Milan, 

Yes I did it, this is the result:

```
PS C:\Users\ajd1\Documents\Code\Happy Birthday> npm install       
up to date, audited 1 package in 503ms
found 0 vulnerabilities
```

Thank you for your support.

**Comment by milanholemans** [comment]
> Local modules not found in

This error means that your node modules are not installed. These modules should install after you run `npm install`. When you open your project, do you see a folder named `node_modules`?

**Comment by hugoabernier** [comment]
@adumith please try the diagnostics steps listed here: https://github.com/pnp/sp-dev-fx-webparts/wiki/Troubleshooting-issues-with-samples

**Comment by adumith** [comment]
> @adumith please try the diagnostics steps listed here: https://github.com/pnp/sp-dev-fx-webparts/wiki/Troubleshooting-issues-with-samples
Hello Hugo, 
Thank you for your support.

I followed the  diagnostics steps, so I got this:

```
PS C:\Users\ajd1\Documents\Code\Happy Birthday> m365 spo cdn get
false

PS C:\Users\ajd1\Documents\Code\Happy Birthday> m365 spfx doctor -o text
 
CLI for Microsoft 365 SharePoint Framework doctor
Verifying configuration of your system for working with the SharePoint Framework

× SharePoint Framework
 
Error: SharePoint Framework not found
```

Maybe that it´s the issue source? How can I fix it?

Thanks in advance,

**Comment by milanholemans** [comment]
> > @adumith please try the diagnostics steps listed here: https://github.com/pnp/sp-dev-fx-webparts/wiki/Troubleshooting-issues-with-samples
> > Hello Hugo,
> > Thank you for your support.
> 
> I followed the diagnostics steps, so I got this:
> 
> ```
> PS C:\Users\ajd1\Documents\Code\Happy Birthday> m365 spo cdn get
> false
> 
> PS C:\Users\ajd1\Documents\Code\Happy Birthday> m365 spfx doctor -o text
>  
> CLI for Microsoft 365 SharePoint Framework doctor
> Verifying configuration of your system for working with the SharePoint Framework
> 
> × SharePoint Framework
>  
> Error: SharePoint Framework not found
> ```
> 
> Maybe that it´s the issue source? How can I fix it?
> 
> Thanks in advance,

Weird, are you sure that you are in the root of the project directory? If I'm not mistaken that error means that it's unable to read the SPFx version of the project.

@hugoabernier since this is a quite old version, should we try to update this project to the latest version?

**Comment by milanholemans** [comment]
> Hi adumith, welcome! The react-birthdays web part is made in SPFx version 1.8.2, this means that you'll need Node.js version 10.x installed on your computer. When this is done, the first thing you need to do is installing the `gulp` package on your computer. This can be achieved by running `npm i gulp -g` in your terminal. Once it is done, navigate to the root of the react-birthdays project. Open your terminal and execute the command `npm i`. This will install all needed packages for this particular project. After this, you can start building the project. Execute following commands:
> 
> ```
> gulp bundle --ship
> gulp package-solution --ship
> ```
> 
> These commands will build and package your project into a single file. You'll find the file in the `./sharepoint/solution` folder. There you'll see a file called `birthdays-anniversary.sppkg`. Upload this file to your SharePoint app catalog and you're done!
> 
> Good luck!

Hi @adumith 

I've updated the web part to the newest SPFx version (1.14.0). PR still has to be reviewed. You can wait for the PR to complete, or download the project from my own repo. You have to do the steps mentioned above all over again. The only difference is that you'll need **Node v14.x** instead of v10.x.

Good luck!
